parsec/backend/vlob.py

```diff
@@ -141,6 +181,68 @@ def __init__(
         self.sequester_services_certificates = sequester_services_certificates
 
 
+async def extract_sequestered_data_and_proceed_webhook(
+    services: Dict[SequesterServiceID, SequesterService],
+    organization_id: OrganizationID,
+    author: DeviceID,
+    encryption_revision: int,
+    vlob_id: VlobID,
+    timestamp: DateTime,
+    sequester_blob: Dict[SequesterServiceID, bytes],
+):
+    # Split storage services and webhook services
+    storage_service_ids = [
+        service.service_id
+        for service in services.values()
+        if service.service_type == SequesterServiceType.STORAGE
+    ]
+    webhook_service_ids = [
+        service.service_id
+        for service in services.values()
+        if service.service_type == SequesterServiceType.WEBHOOK
+    ]
+    for webhook_service_id in webhook_service_ids:
+        service = services[webhook_service_id]
+        sequester_data = sequester_blob[webhook_service_id]
+        if not service.webhook_url:
+            raise VlobSequesterServiceMissingWebhookError(
+                service_id=service.service_id, service_label=service.service_label
+            )
+
+        try:
+            await http_request(
+                url=f"{service.webhook_url}/{organization_id.str}",
+                method="POST",
+                data=sequester_data,
+            )
+        except urllib.error.HTTPError as exc:
+            if exc.code == 400:
+                body = json.loads(exc.read())
+                raise VlobSequesterWebhookServiceRejectedError(
+                    service_id=service.service_id,
+                    service_label=service.service_label,
+                    error=body.get("error", ""),
```

Comment:
Given we provide the response body as an opaque structure, we can instead directly pass `exc.read()`, this way we avoid error handling in json and bytes-to-string decoding

The drawback is we will flood the log if the body is one giant binary blob, but it seems unlikely (and we can provide something like `exc.read()[:2048]` if we really want to avoid this)